sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -88,14 +88,29 @@ void AddCustomerPolicies(HttpPipelinePosition position)
                 {
                     foreach (var policy in options.Policies)
                     {
-                        if (policy.Position == position)
+                        // skip null policies to ensure that calculations for perCallIndex and perRetryIndex are accurate
+                        if (policy.Position == position && policy.Policy != null)
                         {
                             policies.Add(policy.Policy);
                         }
                     }
                 }
             }
 
+            // A helper to ensure that we only add non-null policies to the policies list
+            // This ensures that calculations for perCallIndex and perRetryIndex are accurate
```

Comment:
Do you still need to remove the null policies still?

Reply:
If I understand the question, no, because now they are never added in the first place.

Reply:
So the code at line 126 can be removed?

Reply:
yes! I missed that one, thanks